### PR TITLE
[NOJIRA] Add option popoverZIndex prop to RichSelectList

### PR DESCRIPTION
**Pull request by williamnewman94**
Black Friday is about to begin and our tutor search page is broken:

Screenshot 2024-10-29 at 12 33 30 PM

Tested to see that it forwarded the z-index to the popover properly:

Screenshot 2024-10-29 at 3 45 55 PM

Needed a quick fix because black friday is beginning today. If you have a more robust fix in mind let me know.

### CODE REVIEW

**Comment by somethiiing** [comment]
dont forget to run `pnpm run changeset`

File: packages/syntax-core/src/RichSelect/RichSelectList.tsx [hunk 1]
```diff
@@ -69,6 +69,10 @@ export type RichSelectListProps = RichSelectBoxProps & {
    * @defaultValue white
    */
   color?: "white" | "clear";
+  /**
+   * Z-index of the popover
+   */
+  popoverZIndex?: number;
```
Comment (somethiiing) [inline packages/syntax-core/src/RichSelect/RichSelectList.tsx:75]: can you just call this zIndex?

**Review by somethiiing — APPROVED** [review]
please run changeset to ensure the npm package gets updated + update prop name.

would be really great to update the story with the new prop as well. (you dont have to add a custom story highlighting, but would be great to see it in the prop list)

**Comment by williamnewman94** [comment]
> please run changeset to ensure the npm package gets updated + update prop name.
> 
> would be really great to update the story with the new prop as well. (you dont have to add a custom story highlighting, but would be great to see it in the prop list)

Woops sorry forgot about the changeset!! Added a story as well though I don't think that it is hooked up properly because the change I make to the only param I can change (disabled) does not affect the example.

**Pull request merged**